packages/vue-components/src/ScrollTopButton.vue

```diff
@@ -0,0 +1,140 @@
+<template>
+  <i
+    :class="['scroll-top-button', icon, getIconSize(), 'd-print-none', {'lighten': $data.isLighten}]"
+    :style="iconStyle()"
+    aria-hidden="true"
+    @click="handleScrollTop()"
+  >
+  </i>
+</template>
+
+<script>
+export default {
+  name: 'ScrollTopButton',
+  props: {
+    icon: {
+      default: 'fa fa-arrow-circle-up',
+      type: String,
+    },
+    iconSize: {
+      type: String,
+      default: 'lg',
+    },
+    iconColor: {
+      type: String,
+      default: null,
+    },
+    bottom: {
+      type: String,
+      default: null,
+    },
+    right: {
+      type: String,
+      default: null,
+    },
+  },
+  data() {
+    return {
+      isVisible: false,
+      isLighten: false,
+      timers: {
+        showOrHideButtonTimer: 0,
+        lightenButtonTimer: 0,
+      },
+    };
+  },
+  methods: {
+    initScrollTopButton() {
+      window.addEventListener('scroll', this.promptScrollTopButton);
+    },
+    destroyScrollTopButton() {
+      window.removeEventListener('scroll', this.promptScrollTopButton);
+    },
+    promptScrollTopButton() {
+      this.resetScrollTopButton();
+      this.showOrHideScrollTopButton();
+    },
+    resetScrollTopButton() {
+      clearTimeout(this.timers.showOrHideButtonTimer);
+      clearTimeout(this.timers.lightenButtonTimer);
+      this.isLighten = false;
+    },
+    showOrHideScrollTopButton() {
+      this.timers.showOrHideButtonTimer = setTimeout(() => {
+        if (document.body.scrollTop > 20 || document.documentElement.scrollTop > 20) {
+          this.isVisible = true;
+          this.lightenScrollTopButton();
+        } else {
+          this.isVisible = false;
+        }
+      }, 100);
+    },
+    lightenScrollTopButton() {
+      // lightens the scroll-top-button after 1 second of button inactivity
+      // prevents the button from obscuring the content
+      this.timers.lightenButtonTimer = setTimeout(() => {
+        if (!this.isLighten) {
+          this.isLighten = true;
+        }
+      }, 1000);
+    },
+    iconStyle() {
+      const style = {};
+      style.display = this.isVisible ? 'block' : 'none';
+      if (this.iconColor) {
+        style.color = this.iconColor;
+      }
+      style.bottom = this.bottom ? this.bottom : '2%';
+      style.right = this.right ? this.right : '2%';
+      style.position = 'fixed';
+      return style;
+    },
+    handleScrollTop() {
+      document.body.scrollIntoView({ block: 'start', behavior: 'smooth' });
+    },
+    getIconSize() {
+      return this.iconSize ? `fa-${this.iconSize}` : '';
+    },
+  },
+  mounted() {
+    this.initScrollTopButton();
+  },
+  beforeDestroy() {
+    this.destroyScrollTopButton();
+  },
+};
+</script>
+
+<style scoped>
```

Comment:
```suggestion
<style>
```
In light of #1768  - non-scoped styles have better performance and our class here should be selective enough :)